### PR TITLE
Version improvements

### PR DESCRIPTION
**Pull request by Holzhaus**
- Make version number accessible as QVersionNumber
- Show `git-describe` version in about dialog
- remove git commit count from CMakeLists
- Use git describe for package file name
- Fix quoting when logging build details

### CODE REVIEW

File: src/util/versionstore.cpp [hunk 1]
```diff
@@ -55,21 +71,17 @@ QString VersionStore::applicationName() {
 QString VersionStore::applicationTitle() {
 #ifdef __APPLE__
```
Comment (Holzhaus) [inline src/util/versionstore.cpp:72]: Do we really need the distinction here? I think the application title should always be just `Mixxx`.

File: CMakeLists.txt [hunk 2]
```diff
@@ -2719,9 +2698,9 @@ string(REPLACE "\n" "\n " CPACK_DEBIAN_PACKAGE_DESCRIPTION_MERGED "${CPACK_DEBIA
 # The upstream version must not contain hyphen
 string(REPLACE "-" "~" CPACK_DEBIAN_UPSTREAM_VERSION "${MIXXX_VERSION}")
 set(CPACK_DEBIAN_DEBIAN_VERSION 0ubuntu1)
-set(CPACK_DEBIAN_DEBIAN_VERSION_EXTRA "${GIT_BRANCH}~git${GIT_COMMIT_COUNT}")
+set(CPACK_DEBIAN_DEBIAN_VERSION_EXTRA "${GIT_BRANCH}~git${GIT_DESCRIBE}")
```
Comment (Holzhaus) [inline CMakeLists.txt:2701]: Are there special requirements that might prevent usage of git describe? If so, we should document them as a comment.

**Review by Holzhaus — COMMENTED** [review]
Out of scope for this PR, but will come as a follow up: Use the new QVersionNumber for manual URLs, etc (superseding #3792).

File: CMakeLists.txt (hunk 2 above)
Comment (daschuer) [reply CMakeLists.txt:2701]: Yes there are. New versions numbers have to be "greater" and there are no dashes allowed. 
I have original planned to dive into it and come up with a proposal.

File: src/util/versionstore.cpp (hunk 1 above)
Comment (daschuer) [reply src/util/versionstore.cpp:72]: Yes, I agree. We just need to make sure that this info is inside the about box.

File: CMakeLists.txt (hunk 2 above)
Comment (daschuer) [reply CMakeLists.txt:2701]: How about decouple this question from this PR and keep GIT_COMMIT_COUNT unit we have everything sorted out.

File: CMakeLists.txt [hunk 3]
```diff
@@ -2700,7 +2679,7 @@ list(APPEND CPACK_SOURCE_IGNORE_FILES "/build/")
 list(APPEND CPACK_SOURCE_IGNORE_FILES "/.*_build/")
 list(APPEND CPACK_SOURCE_IGNORE_FILES "/\\\\.sconf_temp/")
 list(APPEND CPACK_SOURCE_IGNORE_FILES "${CMAKE_CURRENT_BINARY_DIR}/")
-set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}-Source")
+set(CPACK_SOURCE_PACKAGE_FILE_NAME "${CPACK_PACKAGE_FILE_NAME}-src")
```
Comment (daschuer) [inline CMakeLists.txt:2682]: what is the reason for this change?

**Comment by daschuer** [comment]
Git describe seems to be failing on our CI:


Error: /home/runner/work/mixxx/mixxx/src/util/versionstore.cpp:41:30: error: Use an empty QLatin1String instead of an empty QStringLiteral [-Wclazy-empty-qstringliteral]
const QString kGitDescribe = QStringLiteral(GIT_DESCRIBE);
                             ^
/usr/include/x86_64-linux-gnu/qt5/QtCore/qstringliteral.h:68:9: note: expanded from macro 'QStringLiteral'
        static const QStaticStringData<Size> qstring_literal = { \
        ^
1 error generated.

**Comment by Swiftb0y** [comment]
> Git describe seems to be failing on our CI:

See: https://github.com/mixxxdj/mixxx/pull/3259/files#diff-680cc13af33346625cdd5b4d0f392ae801b86a3a7b3c60a9f1bb729d4a28e5e1R17

File: CMakeLists.txt (hunk 2 above)
Comment (daschuer) [reply CMakeLists.txt:2701]: The version test is: 
```
dpkg --compare-versions 2.0.0~rc1-0ubuntu1 lt 2.0.0-0ubuntu1 && echo true
```

**Comment by daschuer** [comment]
I have just played a bit with it and it works nice. Thank you. 
During a release, we have a release candidate, where the CMake version has been advanced but we have no tag yet that is acceptable IMHO. I don't think we need to introduce some magic in this case.

**Comment by daschuer** [comment]
How do we trigger the deployment of a release? 
We need to tag the release commit and build it again. Than we need to start the deployment batches. 
On Jenkins that was a build job that was started manually. Does GitHub Actions provide such manual stated jobs?

File: CMakeLists.txt (hunk 2 above)
Comment (daschuer) [reply CMakeLists.txt:2701]: > How about decouple this question from this PR and keep GIT_COMMIT_COUNT unit we have everything sorted out.

Or I can propose a PR against this branch. What do you think?

Comment (Holzhaus) [reply CMakeLists.txt:2701]: I just removed the commit from this PR and we can take care of it in a separate PR.

File: src/util/versionstore.cpp (hunk 1 above)
Comment (Holzhaus) [reply src/util/versionstore.cpp:72]: I removed `applicationTitle` and use `applicationName` instead. The platform information has been improved and is shown in the About dialog now.

File: CMakeLists.txt (hunk 3 above)
Comment (Holzhaus) [reply CMakeLists.txt]: Let's use lowercase file names. This makes it less confusing when dealing with case-(in)sensitive file systems. I changed it to `source` again (I don't actually care if it's `src` or `source`).

**Comment by Holzhaus** [comment]
> How do we trigger the deployment of a release?
> We need to tag the release commit and build it again. Than we need to start the deployment batches.
> On Jenkins that was a build job that was started manually. Does GitHub Actions provide such manual stated jobs?

I'm not sure what you mean, but it seems unrelated to this PR so let's rather discuss this on Zulip.

File: CMakeLists.txt [hunk 4]
```diff
@@ -2685,7 +2687,11 @@ set(CPACK_PACKAGE_INSTALL_DIRECTORY "Mixxx")
 set(CPACK_PACKAGE_EXECUTABLES "mixxx;Mixxx")
 set(CPACK_PACKAGE_ICON "${CMAKE_SOURCE_DIR}/res/images/mixxx_install_logo.bmp")
 set(CPACK_PACKAGE_HOMEPAGE_URL "https://www.mixxx.org/")
-set(CPACK_PACKAGE_FILE_NAME "Mixxx-${GIT_BRANCH}-r${GIT_COMMIT_COUNT}-${GIT_COMMIT_HASH}")
+if(NOT GIT_DESCRIBE)
+  set(CPACK_PACKAGE_FILE_NAME "mixxx-${CMAKE_PROJECT_VERSION}-unknown")
+else()
+  set(CPACK_PACKAGE_FILE_NAME "mixxx-${GIT_DESCRIBE}")
```
Comment (daschuer) [inline CMakeLists.txt:2693]: Do we need to change anything our website because of this?

Comment (Holzhaus) [reply CMakeLists.txt:2693]: No. We don't use the regular package filename anyway (they are renamed by the `deploy.py` script):
https://github.com/mixxxdj/mixxx/blob/main/.github/workflows/build.yml#L329-L331

**Review by daschuer — APPROVED** [review]
LGTM, Thank you very much.
